### ISSUE TITLE
Feature Request: Reshape Arrays

### ISSUE THREAD

**Issue by aaronmswan** [blitzpp/blitz#89]
Reshaping arrays would be a useful feature. Reshape would involve adjusting dimensions such that the product of the dimension does not change, and would not move memory. This would be different than resizeAndPreserve.

It looks like this was considered a while back: https://sourceforge.net/p/blitz/mailman/message/2406484/

**Comment by citibeth** [comment]
See the "reshape()" macro at:

https://github.com/citibeth/ibmisc/blob/develop/slib/ibmisc/blitz.hpp

Maybe some of this stuff should be incorporated directly into Blitz?

**Comment by aaronmswan** [comment]
Thank you for pointing that out. That looks like what I was hoping for.

I think it does make sense for blitz to have this functionality, and maybe the similar ravel function as in the python numpy module.

**Comment by annemenini** [comment]
I agree with @aaronmswan, having features such as reshape, squeeze would be very useful.
